sdk/python/toolchain/pip.go

```diff
@@ -446,11 +446,12 @@ func InstallDependencies(ctx context.Context, cwd, venvDir string, useLanguageVe
 		return nil
 	}
 
-	printmsg("Updating pip, setuptools, and wheel in virtual environment...")
+	printmsg("Updating pip, setuptools, wheel, and debugpy in virtual environment...")
 
 	// activate virtual environment
-
-	err := runPipInstall("updating pip, setuptools, and wheel", "--upgrade", "pip", "setuptools", "wheel")
+	err := runPipInstall(
+		"updating pip, setuptools, wheel, and debugpy",
+		"--upgrade", "pip", "setuptools", "wheel", "debugpy")
```

Comment:
When using Poetry we don't run this code. We don't need these when **using** a pulumi SDK, we only need them for packaging & publishing SDKs. I don't think we should have this code at all, it should probably be added to requirements.txt by the SDK codegen, but it is here now.

To help with the Poetry case, maybe when starting in debug mode, we can first call `toolchain.ListPackages` and check if debugpy is installed. If it is not present, we can tell the user to first install it (`pip install debugpy` or `poetry add debugpy`) and exit with an error.

If we do always want `debugpy` to be available, then we should add it as a dependency of `pulumi` itself.

Reply:
Ah thanks for that!  Yeah `toolchain.ListPackages` sounds like a great idea, and then we can let the user deal with it.  That's probably better than forcing every user to install `debugpy` even if they might not need it.